lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -1616,6 +1618,631 @@ class ConvertAtenAdaptivePoolOp : public OpConversionPattern<OpTy> {
+                highX = b.create<arith::MulFOp>(loc, highX, spatialScaleVal);
+
+                lowY = b.create<arith::SubFOp>(loc, lowY, offset);
+                lowX = b.create<arith::SubFOp>(loc, lowX, offset);
+                highY = b.create<arith::SubFOp>(loc, highY, offset);
+                highX = b.create<arith::SubFOp>(loc, highX, offset);
+
+                // Step 2: Extract region of interest using bounds
+                Value lowY_int = b.create<math::FloorOp>(loc, lowY);
+                Value lowX_int = b.create<math::FloorOp>(loc, lowX);
+                Value highY_int = b.create<math::CeilOp>(loc, highY);
+                Value highX_int = b.create<math::CeilOp>(loc, highX);
+                lowY_int =
+                    b.create<arith::FPToSIOp>(loc, b.getI64Type(), lowY_int);
+                lowX_int =
+                    b.create<arith::FPToSIOp>(loc, b.getI64Type(), lowX_int);
+                highY_int =
+                    b.create<arith::FPToSIOp>(loc, b.getI64Type(), highY_int);
+                highX_int =
+                    b.create<arith::FPToSIOp>(loc, b.getI64Type(), highX_int);
+
+                Value roiHeight =
+                    b.create<arith::SubIOp>(loc, highY_int, lowY_int);
+                Value roiWidth =
+                    b.create<arith::SubIOp>(loc, highX_int, lowX_int);
+
+                SmallVector<Value> roiOffsetVals{zeroAttr, iv1, lowY_int,
+                                                 lowX_int};
+                SmallVector<Value> roiSizeVals{oneAttr, oneAttr, roiHeight,
+                                               roiWidth};
+
+                Value extractRoi = b.create<tensor::ExtractSliceOp>(
+                    loc, input, roiOffsetVals, roiSizeVals, strideVals);
+
+                // Step 3: Perform bilinear interpolation over roi
+                Value roiBinH = b.create<arith::SubOp>(loc, highY, lowY);
+                Value roiBinW = b.create<arith::SubOp>(loc, highX, lowX);
+                Value scaleH = b.create<arith::DivOp>(loc, roiBinH, pooledH);
+                Value scaleW = b.create<arith::DivOp>(loc, roiBinW, pooledW);
+                scaleH = b.create<arith::CeilOp>(loc, scaleH);
+                scaleW = b.create<arith::CeilOp>(loc, scaleW);
+                scaleH = b.create<arith::FPToSIOp>(loc, b.getI64Type(), scaleH);
+                scaleW = b.create<arith::FPToSIOp>(loc, b.getI64Type(), scaleW);
+
+                Value roiSampleHeight =
+                    b.create<arith::MulIOp>(loc, pooledH, scaleH);
+                Value roiSampleWidth =
+                    b.create<arith::MulIOp>(loc, pooledW, scaleW);
+
+                SmallVector<Value> outputSizeIntValues = {roiSampleHeight,
+                                                          roiSampleWidth};
+                SmallVector<Value> dims =
+                    getTensorSizesUntilDim(b, loc, extractRoi, 1);
+                for (unsigned i = 2; i < inputRank; i++) {
```

Comment:
Use pre-increment:
```suggestion
                for (unsigned i = 2; i < inputRank; ++i) {
```
See https://llvm.org/docs/CodingStandards.html#prefer-preincrement

Also in other loops